source/_integrations/zha.markdown

```diff
@@ -33,41 +33,45 @@ There is currently support for the following device types within Home Assistant:
 - Sensor
 - Switch
 
-## ZHA exception and deviation handling
-
-Zigbee devices that deviate from or do not fully conform to the standard specifications set by the [Zigbee Alliance](https://zigbeealliance.org) may require the development of custom [ZHA Device Handlers](https://github.com/dmulcahey/zha-device-handlers) (ZHA custom quirks handler implementation) to for all their functions to work properly with the ZHA integration in Home Assistant. These ZHA Device Handlers for Home Assistant can thus be used to parse custom messages to and from Zigbee devices.
+## Compatible hardware
 
-The custom quirks implementations for zigpy implemented as ZHA Device Handlers for Home Assistant are a similar concept to that of [Hub-connected Device Handlers for the SmartThings Classics platform](https://docs.smartthings.com/en/latest/device-type-developers-guide/) as well as that of [Zigbee-Shepherd Converters as used by Zigbee2mqtt](https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html), meaning they are each virtual representations of a physical device that expose additional functionality that is not provided out-of-the-box by the existing integration between these platforms.
+ZHA integration support using one of the multiple available Zigbee coordinator radio modules that is compatible with [zigpy](https://github.com/zigpy/zigpy). zigpy is an open source Python library implementing a hardware independent Zigbee stack that uses a modular design with separate Python libraries for Zigbee coordinator radio modules from different manufacturers, with each library capable of interfacing with USB and GPIO radio hardware adapters/modules over different UART serial communication protocols. Support for radio libraries in this ZHA integration include; [zigpy-deconz](https://github.com/zigpy/zigpy-deconz) for Dresden-Elektronik deCONZ based radios, [bellows](https://github.com/zigpy/bellows) for Silicon Labs EmberZNet/EZSP based radios, and [zigpy-xbee](https://github.com/zigpy/zigpy-xbee) for XBee based Zigbee radios. There are currently also experimental support for a couple additional radio libraries; [zigpy-cc](https://github.com/sanyatuning/zigpy-cc) for Texas Instruments Z-Stack based radios, and [zigpy-zigate](https://github.com/doudz/zigpy-zigate) for ZiGate based radios.
```

Comment:
Honestly, this paragraph is too big IMHO and contains a lot of irrelevant information for the normal user.

The libraries and packages used, are not relevant for using the integration and might confuse a new user.

Reply:
Maybe, but just will notice that I removed the same links from the hardware list. So you could see it as if the information was just moved, summarized and clarified, so still fewer words all in all. Also, for troubleshooting, for specific Zigbee radio hardware issues the users would really need to submit issues and bug-reports upstream to each radio library repository.

Reply:
For a generic user, they don't know. We have to take into account that our user base will consist of more people without a technical background nowadays.

I think it is fair to expect issue reports on Home Assistant if a user has problems using the integration. We can't expect them to dive into the codebase or to figure out which dependency is effected.

Reply:
Second opinions? My vote goes to keep that paragraph explain how the integration works. Again please note that I also removed the radio library reference from the hardware list so without this the information is not there.

Reply:
A user of Home Assistant should not have to care about which library is used. That is the whole point I'm trying to make.

Added @klaasnicolaas as a second reviewer from the docs team for a second opinion.

Reply:
Please note I also wrote that sentence in order to try to explain to both end-users and potential developers wanting to help that support for even more Zigbee radio hardware from different manufacturers could be added to this integration. Meaning that there is no need to ask/request or make a new integration if you want support for new Zigbee radio hardware from different manufacturers, instead users can ask/request that support for additional Zigbee radio hardware can be added to this same integration even if that hardware comes from different manufacturers.

Reply:
OK I now removed half the paragraph including "Support for radio libraries in this ZHA integration include" and onwards but I then restored radio library links in hardware list.

Reply:
Unresolving the conversation. Awaiting second opinion by @klaasnicolaas, as requested by yourself.